tests/ensemble_evaluator/test_ensemble_builder.py

```diff
@@ -64,3 +67,120 @@ def test_build_ensemble_legacy():
 
     real = ensemble.get_reals()[0]
     assert real.is_active()
+
+
+@pytest.mark.parametrize(
+    "steps,expected,ambiguous",
+    [
+        (
+            [
+                {
+                    "name": "task1",
+                    "inputs": ["parameters"],
+                    "outputs": ["task1_output"],
+                },
+                {
+                    "name": "task2",
+                    "inputs": ["task1_output"],
+                    "outputs": ["task2_output"],
+                },
+                {
+                    "name": "task3",
+                    "inputs": ["task1_output"],
+                    "outputs": ["task3_output"],
+                },
+                {
+                    "name": "task4",
+                    "inputs": [
+                        "task5_output",
+                        "task2_output",
+                        "task6_output",
+                    ],
+                    "outputs": ["task4_output"],
+                },
+                {
+                    "name": "task5",
+                    "inputs": ["task3_output"],
+                    "outputs": ["task5_output"],
+                },
+                {"name": "task6", "inputs": [], "outputs": ["task6_output"]},
+            ],
+            ["task1", "task6", "task2", "task3", "task5", "task4"],
+            [],
+        ),
+        (
+            [
+                {"name": "a", "inputs": [], "outputs": []},
+                {"name": "b", "inputs": [], "outputs": []},
+            ],
+            [],
+            ["a", "b"],
+        ),
+        (
+            [
+                {"name": "a", "inputs": [], "outputs": ["a_out"]},
+                {"name": "b", "inputs": ["a_out"], "outputs": []},
+                {"name": "c", "inputs": [], "outputs": []},
+            ],
+            ["a", "b"],
+            ["c"],
+        ),
+        pytest.param(
+            [
+                {"name": "step0", "inputs": [], "outputs": ["step0_out"]},
+                {"name": "step1", "inputs": ["step0_out"], "outputs": []},
+                {"name": "step1", "inputs": ["step0_out"], "outputs": []},
+            ],
+            [],
+            [],
+            marks=pytest.mark.xfail(raises=ValueError, strict=True),
+        ),
+        pytest.param(
+            [
+                {"name": "step0", "inputs": ["step1_out"], "outputs": ["step0_out"]},
+                {"name": "step1", "inputs": ["step0_out"], "outputs": ["step1_out"]},
+            ],
+            [],
+            [],
+            marks=pytest.mark.xfail(raises=CycleError, strict=True),
+        ),
+    ],
+)
+def test_topological_sort(steps, expected, ambiguous):
```

Comment:
I don't fully understand the concept of an ambiguous step by reading the test and the test data. Could you elaborate? :)

Reply:
Does the provided explanation suffice?